include/seastar/core/preempt.hh

```diff
@@ -38,8 +38,8 @@ struct preemption_monitor {
 
 extern __thread const internal::preemption_monitor* g_need_preempt;
 
-inline bool need_preempt() {
 #ifndef SEASTAR_DEBUG
+inline bool _need_preempt() {
```

Comment:
could move this function into `internal` namespace. like
```diff
diff --git a/include/seastar/core/preempt.hh b/include/seastar/core/preempt.hh
index 722c7831..094ac020 100644
--- a/include/seastar/core/preempt.hh
+++ b/include/seastar/core/preempt.hh
@@ -34,9 +34,7 @@ struct preemption_monitor {
     std::atomic<uint32_t> tail;
 };

-}
-
-extern __thread const internal::preemption_monitor* g_need_preempt;
+extern __thread const preemption_monitor* g_need_preempt;

 inline bool need_preempt() noexcept {
 #ifndef SEASTAR_DEBUG
@@ -49,10 +47,14 @@ inline bool need_preempt() noexcept {
     auto tail = np->tail.load(std::memory_order_relaxed);
     // Possible optimization: read head and tail in a single 64-bit load,
     // and find a funky way to compare the two 32-bit halves.
-    return __builtin_expect(head != tail, false);
+    return head != tail;
 #else
     return true;
 #endif
 }

 }
+
+}
+
+#define need_preempt() __builtin_expect(::seastar::internal::need_preempt(), false)
```

in which, four trivial changes are included: 

* `g_need_preempt` is included in `internal` namespace by including it in the namespace instead of prefixing its name.
* `need_preempt()` is moved into `internal` namespace. as it will be exposed by `need_preempt()` macro.
* i think there is no need to have two `__builtin_expect()` as the outer one is easier for the compiler to consume and to emit more effective code.
* `need_preempt()` is moved out of "seastar" namespace. as macros do not live in a namespace. to define them in a namespace is misleading.

Reply:
Pulled the changes into the last patch.